app/src/main/java/com/example/cm/data/repositories/MeetupRepository.java

```diff
@@ -91,6 +96,40 @@ private void listenToMeetupListChanges() {
                 });
     }
 
+    /**
+     * Get meetup list by list of meetupIds
+     *
+     * @param meetupIds IDs of meetups to retrieve
+     * @return MutableLiveData-List of mutable meetups with ids
+     */
+    public MutableLiveData<List<Meetup>> getMeetupsByIds(List<String> meetupIds) {
+        if (meetupIds == null || meetupIds.isEmpty()) {
+            meetupsForRequestsMLD.postValue(new ArrayList<>());
+            return meetupsForRequestsMLD;
+        }
+
+        List<String> userIdsNoDuplicates = new ArrayList<>(new HashSet<>(meetupIds));
+
+        List<List<String>> subLists = Lists.partition(userIdsNoDuplicates, 10);
+        for (List<String> subList : subLists) {
+            meetupCollection.whereIn(FieldPath.documentId(), subList).addSnapshotListener(executorService,
+                    (value, error) -> {
+                        if (error != null) {
+                            return;
+                        }
+                        if (value != null && !value.isEmpty()) {
+                            List<Meetup> meetups = new ArrayList<>();
+                            for (int i = 0; i < value.getDocuments().size(); i++) {
+                                Meetup meetup = snapshotToMeetup(value.getDocuments().get(i));
+                                meetups.add(meetup);
+                            }
+                            meetupsForRequestsMLD.postValue(meetups);
```

Comment:
Not sure whether just posting a subList works here for getting the complete list. I think we have to use a callback and in the ViewModel fill the list and then send that to the Fragment. Otherwise we might get incomplete lists.

Reply:
ja da stimm ich dir zu, das problem haben wir an mehreren stellen